includes/utils/class-amp-image-dimension-extractor.php

```diff
@@ -2,20 +2,44 @@
 
 class AMP_Image_Dimension_Extractor {
 	static $callbacks_registered = false;
+    const STATUS_FAILED_LAST_ATTEMPT = 'failed';
+    const STATUS_IMAGE_EXTRACTION_FAILED= 'failed';
 
-	static public function extract( $url ) {
+    static public function extract( $urls ) {
```

Comment:
Would you prefer that I add and call an additional method instead, e.g. extract_multi(), leaving the original extract() and its signature intact? It wouldn't be of much use because of the way image dimension extraction works with my changes, but it would still be there. I'd assumed the only client of AMP_Image_Dimension_Extractor would be the AMP-WP plugin since it's a utility included specifically for use by the plugin.

My changes only require that the amp_extract_image_dimensions filter have 1 parameter. Should I change it back to 2, regardless, in case anyone is hooking into the filter from elsewhere?

I had removed the extract_from_attachment_metadata filter for the reasons I explained in my blog post. Should I restore it for the sake of anyone who may have been hooking into it?

I'd appreciate your guidance regarding the above since these are areas where more WordPress specific experience would be especially useful. That said, I like to think I learn quickly!